crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2531,10 +2599,12 @@ impl AuthorityPerEpochStore {
             self.defer_transactions(batch, key, txns)?;
         }
 
-        if randomness_state_updated {
-            if let Some(randomness_manager) = self.randomness_manager.get() {
-                randomness_manager.advance_dkg(batch)?;
+        let mut is_dkg_completed = None;
+        if let Some(randomness_manager) = self.randomness_manager.get() {
+            if randomness_state_updated {
```

Comment:
I'm not sure I follow? This is about whether the state machine was updated, which happens when DKG messages are received. How does the randomness object being in `shared_input_next_versions` relate to that? I think it will always be in there every time, as long as randomness is on and we are still accepting user certs?

Reply:
i think you rewrote this already anyway, but you had ```

                       generate_randomness = true;
                         shared_input_objects.push((
                             SUI_RANDOMNESS_STATE_OBJECT_ID,
                             randomness_obj_initial_shared_version,
                         ));
```

so it looked to me like `generate_randomness` was true iff the random object was in the version map

Reply:
Ah gotcha, yeah will resolve this and new impl ought to be a bit simpler in this regard